src/main/java/com/microsoft/aad/msal4j/IApiParameters.java

```diff
@@ -7,4 +7,6 @@
 
 interface IApiParameters {
     Set<String> scopes();
+
+    ClaimsRequest withClaims();
```

Comment:
'withClaims' is the naming used by the .NET, Android, and (if I understand the syntax) iOS libraries, and it's what the design doc suggests so I'm sure the other libraries will follow it too once they implement it. Plus, having it named a bit different from the existing claims/claimsChallenge fields will probably help inform devs that this has a different purpose from the claims they've already been handling.

It can definitely be used for claims challenge (since they all get merged together before sending), but it is meant to be used in the more general 'request optional/custom claims' scenario. I believe there were some conversations about the design and whether to have separate claims challenge and claims request options or just a single object for all claims scenarios, and the conclusion was to keep them separate: claimsRequest for when a dev has claims they know they want to request, and claims challenge when the token response has some claims challenge and the dev can just copy/paste it into the field.

Reply:
there few options to name setters like setField1(), withField1(), field1() - my point that we use 3rd option. All setters named this way, .Net uses second option.
using different approaches for setter naming in the same SDK is confusing

Reply:
so this is not supposed to be used for claimsChallenge ?